lib/token.js

```diff
@@ -222,6 +259,11 @@ function getDefaultOAuthParams(sdk, oauthOptions) {
     ignoreSignature: sdk.options.ignoreSignature
   };
   util.extend(defaults, oauthOptions);
+
+  if (defaults.grantType === 'authorization_code' && !defaults.codeChallengeMethod) {
+    defaults.codeChallengeMethod = pkce.DEFAULT_CODE_CHALLENGE_METHOD;
```

Comment:
Allowing this value to be configurable is potentially dangerous, say if someone passed in `plain` as the `code_challenge_method`. 

Since you're parsing the `.well-known` response for white-listed values in `prepareOauthParams`, we're able to mitigate this attack. Glad we added that!